### ISSUE TITLE
Compile error - unable to import typeorm

### ISSUE THREAD

**Issue by gnesher** [typeorm/typeorm#3855]
I'm trying to add TypeORM to an existing Cordova / Ionic project with a custom Typescript / Webpack configuration.

When trying to import TypeORM the build fails with the following errors:
```
ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts(11,27)
      TS1005: ';' expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts(11,47)
      TS1005: ';' expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts(11,50)
      TS1109: Expression expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts(11,85)
      TS1005: '(' expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts(11,92)
      TS1005: ',' expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts(11,120)
      TS1005: ',' expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts(11,123)
      TS1109: Expression expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts(11,166)
      TS1005: '(' expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts(11,197)
      TS1005: '(' expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts(11,213)
      TS1005: ')' expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\query-builder\QueryPartialEntity.d.ts(12,1)
      TS1128: Declaration or statement expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts(5,27)
      TS1005: ';' expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts(5,47)
      TS1005: ';' expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts(5,50)
      TS1109: Expression expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts(5,74)
      TS1005: '(' expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts(5,81)
      TS1005: ',' expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts(5,109)
      TS1005: ',' expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts(5,112)
      TS1109: Expression expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts(5,144)
      TS1005: '(' expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts(5,163)
      TS1005: '(' expected.

ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts
[tsl] ERROR in C:\Users\gnesher\work\nortec\locusmap\LocusMap\node_modules\typeorm\common\DeepPartial.d.ts(6,1)
      TS1128: Declaration or statement expected.

```

I've checked the tsconfig file and it seems reasonable - 

```
{
  "compilerOptions": {
    "noImplicitAny": false,
    "noEmitOnError": true,
    "removeComments": false,
    "sourceMap": true,
    "inlineSources": true,
    "experimentalDecorators": true,
    "importHelpers": true,
    "emitDecoratorMetadata": true,
    "noUnusedLocals": true,
    "moduleResolution": "node",
    "module": "commonjs",
    "target": "es5",
    "lib": [ "es5", "dom", "es6"]
  },
  "files": [
    // typings
    "typings/jquery/jquery.d.ts",
    "typings/angularjs/angular.d.ts",
    "typings/angular-ui-router/angular-ui-router.d.ts",
    "typings/angular-local-storage/angular-local-storage.d.ts",
    "typings/leaflet/geojson.d.ts",
    "typings/leaflet/leaflet.d.ts",
    "typings/moment/moment.d.ts",
    "typings/toastr/toastr.d.ts",
    "typings/cryptojs/cryptojs.d.ts",
    "typings/underscore/underscore.d.ts",
    "typings/ionic/ionic.d.ts",
    "typings/ngprogress/ngprogress.d.ts",
    "typings/d3/d3.d.ts",
    "typings/d3/nvd3.d.ts",
    "typings/bcryptjs/index.d.ts",
    
    // Cordova typings
    "typings/cordova/cordova.d.ts",
    "typings/cordova/plugins/Camera.d.ts",
    "typings/cordova/plugins/Device.d.ts",
    "typings/cordova/plugins/FileSystem.d.ts",
    "typings/cordova/plugins/FileTransfer.d.ts",
    "typings/cordova/plugins/imei.d.ts",
    "typings/cordova/plugins/Keyboard.d.ts",
    "typings/cordova/plugins/StatusBar.d.ts",
    "typings/cordova/plugins/NetworkInformation.d.ts",
    "typings/cordova/plugins/Media.d.ts",
    "typings/cordova/plugins/WebSQL.d.ts",
    "typings/cordova/ng-cordova/file.d.ts",
    "typings/cordova/plugins/ChromeSocketsTcp.d.ts",
    "typings/cordova/plugins/BleCentral.d.ts",
    "typings/cordova/plugins/PhotoLibrary.d.ts",
    "typings/cordova/plugins/BackgroundMode.d.ts",
    
    "app/controllers/charts-test.ts",
    "modules/project/modules/pressure-test/components/chart.ts",
    // Startup
    "app.ts"
  ]
}
```

webpack.config.js:
```
const HtmlWebpackPlugin = require("html-webpack-plugin");
const path = require("path");
const ModuleConcatPlugin = require('webpack/lib/optimize/ModuleConcatenationPlugin');
const webpack = require('webpack');

module.exports = {
  entry: "./scripts/app.ts",
  output: {
    path: path.resolve(__dirname),
    filename: "./www/scripts/appBundle.js",
  },
  resolve: {
    extensions: [".tsx", ".ts", ".js"]
  },
  devServer: {
    contentBase: './www'
  },
  optimization: {
    minimize: false
  },
  module: {
    rules: [
      {
        test: /\.tsx?$/,
        exclude: "/node_modules/",
        use: [
          {
            loader: "ts-loader",
            options: {
              configFile: "scripts/tsconfig.json"
            }
          }
        ]
      }
    ]
  },
  plugins: [
    new HtmlWebpackPlugin({
      template: './www/index.ejs'
    }),
    new ModuleConcatPlugin(),
    new webpack.NormalModuleReplacementPlugin(/typeorm$/, function (result) {
      result.request = result.request.replace(/typeorm/, "typeorm/browser");
    })
  ],
  devtool: "source-map"
};
```

Please advise 

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[x] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

Build fails when trying to run webpack - config files included above

**Comment by Kononnable** [comment]
What typescript version are you using?
Can you check this problem was introduced in recent typeorm version or if it some kind of configuration problem? Please try to install some older typeorm version and check if problem still exists.

**Comment by bmjc** [comment]
I ran into the same issue that lead me to this entry.
I managed to verify that the issue has been introduced in 0.2.13

**Comment by Kononnable** [comment]
Pretty old typescript version. Please try with more recent one. You can use npx to test if using more recent version helps. Older version of typescript may not be compatible with type definitions generated by newer versions.
`npx typescript@3.3.3333`

**Comment by gnesher** [comment]
I was using version 3.1 (which comes bundled with Ionic 4 / Angular 7 (I will try to update it though)

**Comment by psyklopz** [comment]
I believe this was corrected in this commit:
https://github.com/typeorm/typeorm/commit/2faf6f6a4592012e581314e51c5b879a5a6604f1

**Comment by psyklopz** [comment]
Duplicates closed #3981

**Comment by imnotjames** [comment]
Typescript `3.3.3333` is required.
